### ISSUE TITLE
Add operation_error hook

### ISSUE THREAD

**Issue by ericem** [Fizzadar/pyinfra#112]
I just finished a deployment using Pyinfra and found it very easy to use. However, one feature I was missing was a way to trap errors for certain operations. I created an operation to send iPhone notifications using Pushover, which worked great, but I also wanted to send notifications when an operation failed. Note: For the deployment I needed to set FAIL_PERCENT=0, so I couldn't use the notification operation after the failed operation since a failure exits the deployment.

**Comment by Fizzadar** [comment]
@ericem I like this - I'm thinking an `on_error` and `on_success` global arguments for all operations which should cover this nicely.

**Comment by Fizzadar** [comment]
I've now implemented this in https://github.com/Fizzadar/pyinfra/commit/7539af7450121a0457a5e839e34b4ef93e134c94, to be released soon in `v0.4`. Looks like:

```py
...

def handle_errors(state, host, op_hash):
    print('Error with host {0} on OP: {1}'.format(host.name, op_hash)

files.put(
    {'Upload some file'},
    'my_file.txt', '/opt/my_file.txt',
    on_error=handle_errors,
)
```
